File: Java/238.Product_of_Array_Except_Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans = new int[nums.length];
        int cur_mult = 1;

        for (int i = 0; i < nums.length; ++i) {
            ans[i] = cur_mult;
            cur_mult *= nums[i];
        }

        cur_mult = 1;
        for (int i = nums.length - 1; i >= 0; --i) {
            ans[i] *= cur_mult;
            cur_mult *= nums[i];
        }
        
        return ans;
    }
}

File: Python3/238.Product_of_Array_Except_Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1 for _ in range(len(nums))]
        cur_mult = 1
        for ind, val in enumerate(nums):
            res[ind] *= cur_mult
            cur_mult *= val

        cur_mult = 1
        for ind, val in enumerate(reversed(nums)):
            res[len(nums) - ind - 1] *= cur_mult
            cur_mult *= val

        return res
